### ISSUE TITLE
esc-key in group create & -edit close the whole stack, not the last dialog

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-desktop#2268]
usually the "escape"-key closes the most recent dialog; in a stack of dialogs, the underlying dialog is show again then (on can regard the "escape"-key as a kind of shortcut to the "cancel" or "close" button).

however, the "escape"-key closes all dialogs in a stack of dialog - eg. if you select "New chat / New group / Add members" and add some members by hand, entering a name and an icon - and finally go to Add member to check things again there and hit "Esc" there - _all_ your work is lost.

(this is an older issue, i checked, it exists also in 1.15.5 so, no worsening in 1.20.0. maybe the issue was introduced with https://github.com/deltachat/deltachat-desktop/issues/1743)

EDIT: the Esc-key works fine at "Settings / Show Classic mail"  and "Settings / Delete message from ..." - here the dialogs are also visually stacked. maybe we can see that approach as a pattern for the other dialogs? at least for future things? Cmp. directly with "Settings / Edit profile" to get the difference.

**Comment by Simon-Laux** [comment]
note: this dialog is technically not stacked, its a single one.

**Comment by r10s** [comment]
> note: this dialog is technically not stacked, its a single one.

okay. however, from the user-perspective, that does not really matter: there is a dialog opening another dialog - otherwise, also the "Cancel" button should always close all dialogs (just to make things clear, i do not propose to do so :)

**Comment by Jikstra** [comment]
We need to kick out the canEscapeKeyClose logic of blueprint and do our own thing. We need to know more about the dialog stack, which we currently don't know (which dialog is at top...)

**Comment by Simon-Laux** [comment]
we need to do keybindings sometime right anyway, maybe this will already come up with improving accessibility, if an dialog is open other keybindings like with chat should be disabled and you should not be able to tap around in the main app while an dialog is open and stuff like that.
But that's probably a bigger thing, for making the dialog better we could just multiple ones on top of each other or add some escape listeners with the prevent default trick/hack as workaround.

**Comment by Jikstra** [comment]
Any ideas how we can do it "right"?

**Comment by farooqkz** [comment]
I can reproduce this:

1. Click "+"
2. Click "New group"
3. Click "add members"
4. Add some members
5. Click "OK"
6. Press Esc

**Comment by Simon-Laux** [comment]
I think this doesn't apply anymore. If anyone disagrees, then please create a new issue.
